package.json

```diff
@@ -37,6 +37,9 @@
     "url": "https://github.com/mikehardy/jetifier/issues"
   },
   "homepage": "https://github.com/mikehardy/jetifier#readme",
+  "dependencies": {
```

Comment:
Is there any other way to find the project workspaces without using any package? I'd like if we keep it dependency-free

Reply:
According to the linked issue, it seemed like in a general sense you could say:
- if "packages" directory exists,
- then foreach subdirectory in packages directory, 
- also scan subdirectory/node_modules for jetification

At the same time the code here looks subtle in a couple ways (trailing slash, defensive coding) https://github.com/viewstools/yarn-workspaces-cra-crna/blob/master/get-yarn-workspaces/index.js - might be nice to not duplicate. @m4tt72 - how do you feel about an optional dependency in the mental direction of "if you use monorepos you are probably use to pain ;-) (sorry monorepo people) so if you want that you have to add an optional dependency for it to work" ? I'm not sure how I feel about it myself actually but you might have ideas?

Reply:
Well, adding a dependency whether it is optional or not is something I tried to avoid from the beginning. There were some cases where I could have used a dependency yet I chose to write a code for that instead... So doing it now will defeat the purpose!

Reply:
@m4tt72 I am not disagreeing on dependencies or not - I think less dependencies are better, and in particular going from 0 to 1 should always be thought about carefully - but I want to mention that you specifically mention prior effort in the context of the current decision and you have to be careful with that as it's a logical fallacy https://en.wikipedia.org/wiki/Sunk_cost#Loss_aversion_and_the_sunk_cost_fallacy - the previous effort was great (previous effort usually is :-) ), but it's in the past. Each new decision should be evaluated independently on it's own merit. For me I don't think this is going to be hard to implement to justify a dependency but I wanted to think about the other side.

Without thinking about previous effort (you already spent the time, it's gone...) - how does an optional dependency actually work? Is it a terrible experience ?

Reply:
I will see if we can get to a simpler solution, just looking for `packages/*` and if they contain `node_modules`. Otherwise, they could be discarded.
Not so experienced with monorepos though, so I don't know if they could be nested in another way. Just saw the issue and thought it would be nice to contribute.
I also like the idea of keeping the project dependency-free, also because we have much more repos that aren't monorepos than the ones that are.

Btw, I'm curious about the idea of using optional dependency, the description on the [docs](http://npm.github.io/using-pkgs-docs/package-json/types/optionaldependencies.html) doesn't seem to match with our case:
>If a dependency can be used, but you would like npm to proceed if it cannot be found or fails to install, then you may put it in the optionalDependencies object.

Reply:
sounds like optional is "terrible" (my word, but basically non-functional) so that's a non-starter, thanks for looking at that. Hopefully then a simple non-dependency solution can do the trick